### DISCUSSION TITLE
Swap left and right channels

### DISCUSSION THREAD

**Discussion by ranperry** [MusicPlayerDaemon/MPD#1601] [General]
Is there a way in MPD to swap the left and right channels?

**Comment by MaxKellermann** [comment]
Yes, with the "route" filter plugin. https://mpd.readthedocs.io/en/stable/plugins.html#route

**Comment by ranperry** [comment]
Thanks. Is there a specific format I should use? Perhaps an example how this is being used as a filter taking into consideration my card is "hw:0,0".

**Comment by ranperry** [comment]
Using Raspberry Pi, I tried this. Am I in the right direction? Documentation is pretty thin...

filter {
    plugin   "route"
    name     "swap stereo channels"
    routes   "0>1,1>0"
}

Assuming 0 is one channel and 1 is another?

**Reply by herman-brix** [reply]
If your mpd audio_output is via alsa, you might consider the following.

The precompiled binary supplied by the Linux distribution I'm using doesn't appear to have the route filter built in so I added the following to /etc/asound.conf

`pcm.lrswap {`
`        type plug`
`        slave.pcm "hw:1,0"`
`        slave.channels 2`
`        ttable.0.1 1`
`        ttable.1.0 1`
`}`

The device then becomes "[pcm.]lrswap" in mpd.conf:

`audio_output {`
`      type    "alsa"`
`       name    "[card name]"`
`       device  "lrswap"`
`}`

Of course in "pcm.[new_device_name]" the new name can be whatever you like.

**Comment by ranperry** [comment]
This is the current deice setting:

audio_output {
        type              "alsa"
        name            "I2SPi"
        device          "hw:CARD=sndrpihifiberry,DEV=0"
}

As per @MaxKellermann I should use the filter option which is native to MPD (I guess). There is not enough info or an example I can use so at this point, I am going by trial and error....
